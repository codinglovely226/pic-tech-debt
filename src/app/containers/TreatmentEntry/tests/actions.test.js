import {
  resetReferralsView,
  resetTreatmentsView,
  updateReferrals,
  updateReferralsViewReferee,
  updateReferralsViewRefereeSite,
  updateReferralsViewSpecialty,
  updateTreatments,
  updateTreatmentsViewContent,
  updateTreatmentsViewDisposition,
  updateTreatmentsViewTreatments,
} from '../actions/case';

describe('Treatments action creators', () => {
  it('resetReferralsView', () => {
    expect(resetReferralsView()).toMatchSnapshot();
  });
  it('resetTreatmentsView', () => {
    expect(resetTreatmentsView()).toMatchSnapshot();
  });
  it('updateReferrals', () => {
    expect(updateReferrals(1, 2, 'Test Referee', 'Test Specialty')).toMatchSnapshot();
  });
  it('updateReferralsViewReferee', () => {
    expect(updateReferralsViewReferee('Test Referee')).toMatchSnapshot();
  });
  it('updateReferralsViewRefereeSite', () => {
    expect(updateReferralsViewRefereeSite(1)).toMatchSnapshot();
  });
  it('updateReferralsViewSpecialty', () => {
    expect(updateReferralsViewSpecialty('Test Specialty')).toMatchSnapshot();
  });
  it('updateTreatments', () => {
    expect(updateTreatments(1, 2, 'Test Disposition', ['Treatment 1'], { name: 'Test SPI' })).toMatchSnapshot();
  });
  it('updateTreatmentsViewContent', () => {
    expect(updateTreatmentsViewContent('Test Content')).toMatchSnapshot();
  });
  it('updateTreatmentsViewDisposition', () => {
    expect(updateTreatmentsViewDisposition('Test Disposition')).toMatchSnapshot();
  });
  it('updateTreatmentsViewTreatments', () => {
    expect(updateTreatmentsViewTreatments(['Test Treatment 2'])).toMatchSnapshot();
  });
});
