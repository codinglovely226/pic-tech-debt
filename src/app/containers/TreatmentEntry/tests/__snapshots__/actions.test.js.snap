exports[`Treatments action creators resetReferralsView 1`] = `
Object {
  "type": "pic/TreatmentEntry/RESET_REFERRALS_VIEW",
}
`;

exports[`Treatments action creators resetTreatmentsView 1`] = `
Object {
  "type": "pic/TreatmentEntry/RESET_TREATMENTS_VIEW",
}
`;

exports[`Treatments action creators updateReferrals 1`] = `
Object {
  "id": 1,
  "referee": 2,
  "refereeSite": NaN,
  "specialty": "Test Referee",
  "type": "pic/TreatmentEntry/UPDATE_REFERRALS",
}
`;

exports[`Treatments action creators updateReferralsViewReferee 1`] = `
Object {
  "referee": "Test Referee",
  "type": "pic/TreatmentEntry/UPDATE_REFERRALS_VIEW_REFEREE",
}
`;

exports[`Treatments action creators updateReferralsViewRefereeSite 1`] = `
Object {
  "refereeSite": 1,
  "type": "pic/TreatmentEntry/UPDATE_REFERRALS_VIEW_REFEREE_SITE",
}
`;

exports[`Treatments action creators updateReferralsViewSpecialty 1`] = `
Object {
  "specialty": "Test Specialty",
  "type": "pic/TreatmentEntry/UPDATE_REFERRALS_VIEW_SPECIALTY",
}
`;

exports[`Treatments action creators updateTreatments 1`] = `
Object {
  "disposition": "Test Disposition",
  "id": 1,
  "noteId": 2,
  "spi": Object {
    "name": "Test SPI",
  },
  "treatments": Array [
    "Treatment 1",
  ],
  "type": "pic/TreatmentEntry/UPDATE_TREATMENTS",
}
`;

exports[`Treatments action creators updateTreatmentsViewContent 1`] = `
Object {
  "content": "Test Content",
  "type": "pic/TreatmentEntry/UPDATE_TREATMENTS_VIEW_CONTENT",
}
`;

exports[`Treatments action creators updateTreatmentsViewDisposition 1`] = `
Object {
  "disposition": "Test Disposition",
  "type": "pic/TreatmentEntry/UPDATE_TREATMENTS_VIEW_DISPOSITION",
}
`;

exports[`Treatments action creators updateTreatmentsViewTreatments 1`] = `
Object {
  "treatmentList": Array [
    "Test Treatment 2",
  ],
  "type": "pic/TreatmentEntry/UPDATE_TREATMENTS_VIEW_TREATMENTS",
}
`;
